cmake/public/utils.cmake

```diff
@@ -353,16 +353,28 @@ function(torch_compile_options libname)
       set(MSVC_DEBINFO_OPTION "/Zi")
     endif()
 
+    set(MSVC_APPEND_OPTION)
+
     if(${MSVC_TOOLSET_VERSION} GREATER_EQUAL 142)
       # Add /permissive- flag for conformance mode to the compiler.
       # This will force more strict check to the code standard.
       # 1. From MS official doc: https://learn.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=msvc-170#remarks
       #    By default, the /permissive- option is set in new projects created by Visual Studio 2017 version 15.5 and later versions.
       #    We set the /permissive- flag from VS 2019 (MSVC_TOOLSET_VERSION 142) to avoid compiling issues for old toolkit.
       # 2. For MSVC VERSION: https://cmake.org/cmake/help/latest/variable/MSVC_TOOLSET_VERSION.html
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /permissive-" PARENT_SCOPE)
+      target_compile_options(${libname} PUBLIC $<$<COMPILE_LANGUAGE:CXX>:/permissive->)
+      message("torch_compile_options: ${libname} append /permissive-")
+    endif()
+
+    if(${MSVC_TOOLSET_VERSION} GREATER_EQUAL 143)
```

Comment:
if(${MSVC_TOOLSET_VERSION} GREATER_EQUAL 143 AND NOT CXX_AVX512_FOUND)

Reply:
If a Windows host supports AVX512, we shouldn't disable it.

Reply:
you'd better try to know about the different between compiler ISA check and runtime ISA check.